### PR TITLE
[test-only] ApiTest. test for password policy

### PR DESCRIPTION
**Pull request by ScharfViktor**
- [x] added api test for password policy feature. see here https://github.com/owncloud/ocis/pull/7195
- [x] added method when we use more than one env

### CODE REVIEW

**Comment by ScharfViktor** [comment]
sometimes Given step `user "Alice" has created a public link share with settings` failed. example https://drone.owncloud.com/owncloud/ocis/26528/52/6

File: tests/acceptance/features/apiGraph/enforcePasswordPublicLink.feature [hunk 1]
```diff
@@ -60,3 +60,155 @@ Feature: enforce password on public link
       | ocs-api-version | ocs-code |
       | 1               | 100      |
       | 2               | 200      |
+
+
+  Scenario Outline: creating the public link password in accordance with the password policy
```
Comment (saw-jan) [inline tests/acceptance/features/apiGraph/enforcePasswordPublicLink.feature:65]:
```suggestion
  Scenario Outline: create a public link with a password in accordance with the password policy
```

File: tests/acceptance/features/apiGraph/enforcePasswordPublicLink.feature [hunk 2]
```diff
@@ -60,3 +60,155 @@ Feature: enforce password on public link
...
+    Given using OCS API version "<ocs-api-version>"
+    And the config has been set to these config and value pairs:
```
Comment (saw-jan) [inline tests/acceptance/features/apiGraph/enforcePasswordPublicLink.feature:67]:
```suggestion
    And the following configs have been set:
```

File: tests/acceptance/features/apiGraph/enforcePasswordPublicLink.feature [hunk 3]
```diff
@@ -60,3 +60,155 @@ Feature: enforce password on public link
...
+    And the public should not be able to download file "/textfile.txt" from inside the last public link shared folder using the new public WebDAV API without a password
+    And the public should not be able to download file "/textfile.txt" from inside the last public link shared folder using the new public WebDAV API with password "wrong pass"
+    But the public should be able to download file "/textfile.txt" from inside the last public link shared folder using the new public WebDAV API with password "3s:5WW9uE5h=A"
+    Examples:
+      | ocs-api-version | ocs-code |
+      | 1               | 100      |
+      | 2               | 200      |
+
+
+  Scenario Outline: creating a public link password that does not comply with the password policy
```
Comment (saw-jan) [inline tests/acceptance/features/apiGraph/enforcePasswordPublicLink.feature:90]:
```suggestion
  Scenario Outline: create a public link with a password that does not comply with the password policy
```

File: tests/acceptance/features/apiGraph/enforcePasswordPublicLink.feature (hunk 1 above)
Comment (saw-jan) [inline tests/acceptance/features/apiGraph/enforcePasswordPublicLink.feature:65]:
```suggestion
  Scenario Outline: create a public link with a password in accordance with the password policy
```

File: tests/acceptance/features/apiGraph/enforcePasswordPublicLink.feature (hunk 3 above)
Comment (saw-jan) [inline tests/acceptance/features/apiGraph/enforcePasswordPublicLink.feature:90]:
```suggestion
  Scenario Outline: create a public link with a password that does not comply with the password policy
```

File: tests/acceptance/features/apiGraph/enforcePasswordPublicLink.feature [hunk 4]
```diff
@@ -60,3 +60,155 @@ Feature: enforce password on public link
...
+    Then the HTTP status code should be "<http-code>"
+    And the OCS status code should be "400"
+    And the OCS status message should be "password validation failed"
+    Examples:
+      | ocs-api-version | http-code |
+      | 1               | 200       |
+      | 2               | 400       |
+
+
+  Scenario Outline: updating the public link password in accordance with the password policy
```
Comment (saw-jan) [inline tests/acceptance/features/apiGraph/enforcePasswordPublicLink.feature:112]:
```suggestion
  Scenario Outline: update a public link with a password in accordance with the password policy
```

File: tests/acceptance/features/apiGraph/enforcePasswordPublicLink.feature [hunk 5]
```diff
@@ -60,3 +60,155 @@ Feature: enforce password on public link
...
+    And the public should not be able to download file "/textfile.txt" from inside the last public link shared folder using the new public WebDAV API without a password
+    And the public should not be able to download file "/textfile.txt" from inside the last public link shared folder using the new public WebDAV API with password "wrong pass"
+    But the public should be able to download file "/textfile.txt" from inside the last public link shared folder using the new public WebDAV API with password "6a0Q;A3 +i^m["
+    Examples:
+      | ocs-api-version | ocs-code |
+      | 1               | 100      |
+      | 2               | 200      |
+
+
+  Scenario Outline: updating a public link password that does not comply with the password policy
```
Comment (saw-jan) [inline tests/acceptance/features/apiGraph/enforcePasswordPublicLink.feature:139]:
```suggestion
  Scenario Outline: update a public link with a password that does not comply with the password policy
```

File: tests/acceptance/features/apiGraph/enforcePasswordPublicLink.feature [hunk 6]
```diff
@@ -60,3 +60,155 @@ Feature: enforce password on public link
...
+      at least 1 numbers are required
+      at least 2 special characters are required.  !"#$%&'()*+,-./:;<=>?@[\]^_`{|}~
+      """
+    Examples:
+      | ocs-api-version | http-code |
+      | 1               | 200       |
+      | 2               | 400       |
+
+
+  Scenario Outline: updating the public link password in accordance with the password policy. positive cases
```
Comment (saw-jan) [inline tests/acceptance/features/apiGraph/enforcePasswordPublicLink.feature:170]:
```suggestion
  Scenario Outline: update a public link with a password in accordance with the password policy (valid cases)
```

File: tests/acceptance/features/apiGraph/enforcePasswordPublicLink.feature [hunk 7]
```diff
@@ -60,3 +60,155 @@ Feature: enforce password on public link
...
+      | FRONTEND_PASSWORD_POLICIES_MIN_CHARACTERS           | 10           | with space                        |
+      | FRONTEND_PASSWORD_POLICIES_MIN_LOWERCASE_CHARACTERS | 3            | Test                              |
+      | FRONTEND_PASSWORD_POLICIES_MIN_UPPERCASE_CHARACTERS | 3            | TeST                              |
+      | FRONTEND_PASSWORD_POLICIES_MIN_DIGITS               | 1            | test1                             |
+      | FRONTEND_PASSWORD_POLICIES_MIN_SPECIAL_CHARACTERS   | 1            | test pass                         |
+      | FRONTEND_PASSWORD_POLICIES_MIN_SPECIAL_CHARACTERS   | 33           | ! #$%&'()*+,-./:;<=>?@[\]^_`{  }~ |
+      | FRONTEND_PASSWORD_POLICIES_MIN_SPECIAL_CHARACTERS   | 5            | sameCharacterShouldWork!!!!!      |
+
+
+  Scenario Outline: updating the public link password in accordance with the password policy. negotive cases
```
Comment (saw-jan) [inline tests/acceptance/features/apiGraph/enforcePasswordPublicLink.feature:197]:
```suggestion
  Scenario Outline: update a public link with a password that does not comply with the password policy (invalid cases)
```

File: tests/acceptance/features/apiGraph/enforcePasswordPublicLink.feature (hunk 2 above)
Comment (saw-jan) [inline tests/acceptance/features/apiGraph/enforcePasswordPublicLink.feature:67]:
```suggestion
    And the following configs have been set:
```

File: tests/acceptance/features/apiGraph/enforcePasswordPublicLink.feature [hunk 8]
```diff
@@ -60,3 +60,155 @@ Feature: enforce password on public link
...
+    But the public should be able to download file "/textfile.txt" from inside the last public link shared folder using the new public WebDAV API with password "3s:5WW9uE5h=A"
+    Examples:
+      | ocs-api-version | ocs-code |
+      | 1               | 100      |
+      | 2               | 200      |
+
+
+  Scenario Outline: creating a public link password that does not comply with the password policy
+    Given using OCS API version "<ocs-api-version>"
+    And the config has been set to these config and value pairs:
```
Comment (saw-jan) [inline tests/acceptance/features/apiGraph/enforcePasswordPublicLink.feature:92]:
```suggestion
    And the following configs have been set:
```

File: tests/acceptance/features/apiGraph/enforcePasswordPublicLink.feature [hunk 9]
```diff
@@ -60,3 +60,155 @@ Feature: enforce password on public link
...
+    And the OCS status message should be "password validation failed"
+    Examples:
+      | ocs-api-version | http-code |
+      | 1               | 200       |
+      | 2               | 400       |
+
+
+  Scenario Outline: updating the public link password in accordance with the password policy
+    Given using OCS API version "<ocs-api-version>"
+    And the config has been set to these config and value pairs:
```
Comment (saw-jan) [inline tests/acceptance/features/apiGraph/enforcePasswordPublicLink.feature:114]:
```suggestion
    And the following configs have been set:
```

File: tests/acceptance/features/apiGraph/enforcePasswordPublicLink.feature [hunk 10]
```diff
@@ -60,3 +60,155 @@ Feature: enforce password on public link
...
+    But the public should be able to download file "/textfile.txt" from inside the last public link shared folder using the new public WebDAV API with password "6a0Q;A3 +i^m["
+    Examples:
+      | ocs-api-version | ocs-code |
+      | 1               | 100      |
+      | 2               | 200      |
+
+
+  Scenario Outline: updating a public link password that does not comply with the password policy
+    Given using OCS API version "<ocs-api-version>"
+    And the config has been set to these config and value pairs:
```
Comment (saw-jan) [inline tests/acceptance/features/apiGraph/enforcePasswordPublicLink.feature:141]:
```suggestion
    And the following configs have been set:
```

**Comment by saw-jan** [comment]
CI failure must be flaky one. I have restarted it

**Comment by ScharfViktor** [comment]
After debugging and some experimentation with CI, it turned out that flaky happens after restarting the server (using ociswrapper). 

The cases where we can get flaky:
- Given steps (creating users, files) - usually steps from `Background`  
- setting new envs (using ociswrapper)
- given steps (creating link) - here I get flaky. Sometimes test doesn't find users personal space or files
- test action (When, Then steps)

I changed line order like: 
- setting new envs (using ociswrapper)
- given steps
- test action (When, Then steps)

**Comment by ScharfViktor** [comment]
I ran CI a few times just to be sure. All the runs were green.

**Review by saw-jan — APPROVED** [review]
:+1:

**Pull request merged**